File: chat_tester.py
#!/usr/bin/env python3
import pexpect
import sys
import time

def test_conversation():
    print("🧪 OCLI Conversational Test")
    print("=" * 60)
    print()
    
    # Start OCLI
    print("🚀 Starting OCLI...")
    child = pexpect.spawn("/home/drew/projects/ollama-ocli/target/release/ocli", 
                          encoding="utf-8", timeout=30)
    child.logfile = sys.stdout
    
    # Wait for startup
    time.sleep(3)
    
    # Test 1: Conversational question
    print("\n" + "=" * 60)
    print("🔵 TEST 1: Conversational AI")
    print("=" * 60)
    print("📥 Sending: What is 2+2?")
    child.sendline("What is 2+2?")
    time.sleep(8)
    
    # Test 2: Slash command
    print("\n" + "=" * 60)
    print("🔵 TEST 2: Slash Command")
    print("=" * 60)
    print("📥 Sending: /write-direct /tmp/test.txt Hello")
    child.sendline("/write-direct /tmp/test.txt Hello")
    time.sleep(2)
    
    # Exit
    print("\n" + "=" * 60)
    print("🔵 Exiting OCLI")
    print("=" * 60)
    child.sendline("exit")
    child.expect(pexpect.EOF, timeout=5)
    
    print("\n✅ Test complete!")

if __name__ == "__main__":
    test_conversation()

File: chat_tester.go
package main

import (
	"bufio"
	"fmt"
	"io"
	"os/exec"
	"time"
)

func main() {
	fmt.Println("🧪 OCLI Conversational Test")
	fmt.Println("============================\n")

	cmd := exec.Command("/home/drew/projects/ollama-ocli/target/release/ocli")
	
	stdin, _ := cmd.StdinPipe()
	stdout, _ := cmd.StdoutPipe()
	stderr, _ := cmd.StderrPipe()
	
	cmd.Start()
	
	// Read output
	go func() {
		scanner := bufio.NewScanner(stdout)
		for scanner.Scan() {
			fmt.Println("📤 " + scanner.Text())
		}
	}()
	
	go func() {
		scanner := bufio.NewScanner(stderr)
		for scanner.Scan() {
			fmt.Println("⚠️  " + scanner.Text())
		}
	}()
	
	time.Sleep(2 * time.Second)
	
	// Test conversation
	tests := []struct {
		input   string
		wait    int
		desc    string
	}{
		{"What is 2+2?", 8, "Simple math question"},
		{"/write-direct /tmp/test.txt Hello", 2, "Slash command"},
		{"exit", 1, "Exit"},
	}
	
	for i, test := range tests {
		fmt.Printf("\n🔵 Test %d: %s\n", i+1, test.desc)
		fmt.Printf("📥 Input: %s\n", test.input)
		
		io.WriteString(stdin, test.input+"\n")
		time.Sleep(time.Duration(test.wait) * time.Second)
	}
	
	stdin.Close()
	cmd.Wait()
	
	fmt.Println("\n✅ Complete!")
}
